src/main/resources/org/opentripplanner/apis/gtfs/schema.graphqls

```diff
@@ -1640,30 +1645,32 @@ type RideHailingEstimate {
     productName: String
 }
 
-type Leg {
-    """
-    The date and time when this leg begins. Format: Unix timestamp in milliseconds.
-    """
-    startTime: Long
+"""
+An ISO-8601-formatted datetime with offset, i.e. `2023-06-13T14:30+03:00` for 2:30pm on June 13th 2023 at Helsinki's offset from UTC at that time.
 
-    """
-    The date and time when this leg ends. Format: Unix timestamp in milliseconds.
-    """
-    endTime: Long
+ISO-8601 allows many different formats but OTP will only return the profile specified in RFC3339.
+"""
+scalar OffsetDateTime @specifiedBy(url: "https://www.rfcreader.com/#rfc3339")
 
+type RealtimeEstimate {
+    time: OffsetDateTime
     """
-    For transit leg, the offset from the scheduled departure time of the boarding
-    stop in this leg, i.e. scheduled time of departure at boarding stop =
-    `startTime - departureDelay`
-    """
-    departureDelay: Int
+    The delay or "earliness" of the vehicle at a certain place.
 
+    If the vehicle is early then this is a negative duration.
     """
-    For transit leg, the offset from the scheduled arrival time of the alighting
-    stop in this leg, i.e. scheduled time of arrival at alighting stop = `endTime
-    - arrivalDelay`
-    """
-    arrivalDelay: Int
+    scheduleOffset: Duration
+}
+
+type LegTimes {
```

Comment:
- Call?

Reply:
My suggestion is - The GTFS RT spec uses `scheduledTime`, `predictedTime` and `delay` either as field names or in he description.
 
```
type LegTime {
  scheduled : IsoTime
  predicted : IsoTime
  delay : Duration
}
```

Why not call? A Call has a relationship to a Quay(Stop) and a ServiceJourney(Trip) (in Siri). So it does not fit with a leg, since a leg may or may not depart/arrive at a stop and may/or may not have a Trip.

Reply:
See https://gtfs.org/realtime/best-practices/#tripupdate for a description of delay

Reply:
The TimeEvent for a trip arriving at a stop has more information assosiated with it, so I think it is good to use another type for that; hence `LegTime` here, another option is `LegTimeEvent`. I think event works well, but to many people a event is something happening at a specific time (I agree) at a specific place (I do not agree so much). 

The type should have a name in singular form `LegTime` not `LegTimes` - it is not a collection of `LegTime` - Strictly it is just one time as well, just different "versions" of it.

Reply:
I quite like that the the realtime information is in a separate object because there are a lot of cases where you don't have realtime information and `predicted: null` quite nicely captures that.

If we go down the route of a single object I think I would like to use an enum instead of a boolean `realtimeExists`.

Lets discuss all of this in the dev meeting.